Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2011 - 2018 Schibsted Products & Technology AS.
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import Foundation
+
+private extension DateFormatter {
+    static let local: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+        return formatter
+    }()
+}
+
+private enum Keys {
+    static let agreementsData = "user-agreements-data"
+    enum Data {
+        static let userID = "user-id"
+        static let date = "date"
+        static let agreements = "agreements"
+    }
+}
+
+extension SDKConfiguration {
+    /// How agreements are cached
+    public struct AgreementsCache {
+        static let `default` = AgreementsCache(
+            isOn: true,
+            minDays: 1,
+            maxDays: 7
+        )
+
+        /// Whether caching is enabled or not
+        public let isOn: Bool
+        /// Minimum number of days agreements status should be cached
+        public let minDays: UInt32
+        /// Maximum number of days agreements status should be cached
+        public let maxDays: UInt32
+
+        func load(forUserID userID: String) -> Agreements? {
+            guard self.isOn, let data = Settings.value(forKey: Keys.agreementsData) as? JSONObject else {
+                return nil
+            }
+            do {
+                let storedID = try data.string(for: Keys.Data.userID)
+                guard storedID == userID else {
```

Comment:
Any reason to not combine the `guards` here? One test on each line?

Reply:
Just readability. Also might need try? but then I can't log the throw if it throws

Reply:
The try-catch is ok.. Was thinking more about the 3 guards :)